### ISSUE TITLE
[6.X] spell system

### ISSUE THREAD

**Issue by soulfrost** [TrinityCore/TrinityCore#15754]
Hi, Trinitycore Rev: https://github.com/TrinityCore/TrinityCore/commit/935c93e099f534f701ff8787204ce19be2ed2df2

value scaling for spells which casted from items not working, f.e. http://www.wowhead.com/item=118876/lucky-double-sided-coin give not correct stat value

**Comment by soulfrost** [comment]
p.s.
in 
`` int32 SpellEffectInfo::CalcValue( 

value = GetRandomPropertyPoints(_spellInfo->Scaling.ScalesFromItemLevel, ITEM_QUALITY_RARE, INVTYPE_CHEST, 0);
```
should not we use here m_castitem->GetItemLevel() and use functions to get quality/invType ?

**Comment by Shauren** [comment]
Not quite, what you found in the code is exactly how client calculates the value for spell tooltip so I would need to find more about this.

**Comment by Shauren** [comment]
Fixed in d5fb18d3c3a557acaf60a39be4ad5444969dd663
